src/test/java/org/opentripplanner/middleware/utils/ItineraryUtilsTest.java

```diff
@@ -0,0 +1,360 @@
+
+        Map<ZonedDateTime, String> queriesByDate = ItineraryUtils.getQueriesFromDates(trip.parseQueryParams(), newZonedDateTimes);
+        Assertions.assertEquals(newDates.size(), queriesByDate.size());
+
+        for (ZonedDateTime zonedDateTime : newZonedDateTimes) {
+            MonitoredTrip newTrip = new MonitoredTrip();
+            newTrip.queryParams = queriesByDate.get(zonedDateTime);
+
+            Map<String, String> newParams = newTrip.parseQueryParams();
+            Assertions.assertEquals(zonedDateTime.format(DateTimeFormatter.ofPattern(DEFAULT_DATE_FORMAT_PATTERN)),
+                newParams.get(DATE_PARAM));
+        }
+    }
+
+    /**
+     * Check the computation of the dates corresponding to the monitored days,
+     * for which we want to check itinerary existence.
+     */
+    @ParameterizedTest
+    @MethodSource("createGetDatesTestCases")
+    public void testGetDatesToCheckItineraryExistence(Set<ZonedDateTime> testDates, boolean checkAllDays) throws URISyntaxException {
+        MonitoredTrip trip = makeTestTrip();
+        Set<ZonedDateTime> datesToCheck = ItineraryUtils.getDatesToCheckItineraryExistence(trip, checkAllDays);
+        Assertions.assertEquals(testDates, datesToCheck);
+    }
+
+    private static Stream<Arguments> createGetDatesTestCases() {
+        // Each list includes dates to be monitored in a 7-day window starting from the query date.
+        return Stream.of(
+            // Dates solely based on monitored days (see the trip variable in the corresponding test).
+            Arguments.of(datesToZonedDateTimes(
+                List.of(QUERY_DATE /* Thursday */, "2020-08-15", "2020-08-16", "2020-08-17", "2020-08-18")
+            ), false)
+
+            // If we forceAllDays to ItineraryUtils.getDatesToCheckItineraryExistence,
+            // it should return all dates in the 7-day window regardless of the ones set in the monitored trip.
+            //new GetDatesTestCase(List.of(QUERY_DATE /* Thursday */, "2020-08-14", "2020-08-15", "2020-08-16", "2020-08-17", "2020-08-18", "2020-08-19"))
+
+        );
+    }
+
+    /**
+     * Check that the ignoreRealtime query parameter is set to true
+     * regardless of whether it was originally missing or false.
+     */
+    @Test
+    public void testAddIgnoreRealtimeParam() throws URISyntaxException {
+        String queryWithRealtimeParam = BASE_QUERY + "&" + IGNORE_REALTIME_UPDATES_PARAM + "=false";
+        List<String> queries = List.of(BASE_QUERY, queryWithRealtimeParam);
+
+        for (String query : queries) {
+            MonitoredTrip trip = new MonitoredTrip();
+            trip.queryParams = query;
+            Map<String, String> params = ItineraryUtils.excludeRealtime(trip.parseQueryParams());
+            Assertions.assertEquals("true", params.get(IGNORE_REALTIME_UPDATES_PARAM));
+        }
+    }
+
+    /**
+     * Check whether certain itineraries match.
```

Comment:
I think it would make sense to have unit tests for leg matching (and perhaps stop matching too).

Reply:
I think that'd be a bit too detailed. If anything more Itinerary matching test cases could be created to test differences with legs. But I'm a bit hesitant to get too detailed because I believe we'll eventually need to shift away to discerning more monitored trip matching and existence using OTP endpoints thus removing the need for this code and tests.